packages/sdk/src/wallet/core/wallets/smart/default/DefaultSmartWallet.ts

```diff
@@ -307,6 +307,53 @@ export class DefaultSmartWallet extends SmartWallet {
     return signerIndex
   }
 
+  /**
+   * Remove an existing signer from the smart wallet
+   * @description Removes a signer (EOA address or WebAuthn public key) from the
+   * smart wallet contract. If `signerIndex` is not provided, the method resolves
+   * it via {@link findSignerIndex}. The removal is executed via {@link sendBatch}
+   * by calling the contract function `removeOwnerAtIndex(index, signerBytes)`.
+   * Returns the ERC-4337 UserOperation receipt on success.
+   * @param signer - Signer to remove: EOA address or `WebAuthnAccount`
+   * @param chainId - Target chain on which the smart wallet operates
+   * @param signerIndex - Optional known on-chain index of the signer (skips lookup when provided)
+   * @returns Promise resolving to the UserOperation receipt for the removal
+   * @throws Error if the signer index cannot be found or the removal operation fails
+   */
+  async removeSigner(
+    signer: Signer,
+    chainId: SupportedChainId,
+    signerIndex?: number,
+  ): Promise<WaitForUserOperationReceiptReturnType> {
+    const resolvedSignerIndex =
+      signerIndex ?? (await this.findSignerIndex(signer, chainId))
+    if (resolvedSignerIndex === -1) {
+      throw new Error('failed to find signer index')
+    }
+    const signerBytes = formatPublicKey(signer)
+    const calls = [
+      {
+        to: this.address,
+        data: encodeFunctionData({
+          abi: smartWalletAbi,
+          functionName: 'removeOwnerAtIndex',
+          args: [BigInt(resolvedSignerIndex), signerBytes] as const,
```

Comment:
🤔 What happens if I provide the wrong index? I see the contract takes both.  Since the signer address is required, it feels safer to remove the ability for them to even include the index and just always look it up to be certain. Thoughts?